### PR TITLE
feat: added 2 0 indexers

### PR DESCRIPTION
**Pull request by maximilianwerk**
Showcasing 2.0 indexers, including trivial(!) compounding.

### CODE REVIEW

File: indexers/dbms/BinaryPbIndexer/binarypb.py [hunk 1]
```diff
@@ -0,0 +1,239 @@
+__copyright__ = "Copyright (c) 2021 Jina AI Limited. All rights reserved."
+__license__ = "Apache-2.0"
+__copyright__ = "Copyright (c) 2020 Jina AI Limited. All rights reserved."
```
Comment (cristianmtr) [inline indexers/dbms/BinaryPbIndexer/binarypb.py:3]: not sure what happened here

File: indexers/dbms/BinaryPbIndexer/__init__.py [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.close()
+
+    @property
+    def is_exist(self) -> bool:
+        """
+        Check if the database is exist or not
```
Comment (cristianmtr) [inline indexers/dbms/BinaryPbIndexer/__init__.py:92]: phrasing is off. I know it was there before, but now it's the time to change it :)

```suggestion
        Check if the database exists or not
```

File: indexers/dbms/BinaryPbIndexer/__init__.py [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+
+        :return: true if the absolute index path exists, else false
+        """
+        return os.path.exists(self.index_abspath)
+
+    @cached_property
+    def write_handler(self):
+        """A writable and indexable object, could be dict, map, list, numpy array etc.
```
Comment (cristianmtr) [inline indexers/dbms/BinaryPbIndexer/__init__.py:100]: docstring does not reflect implementation

File: indexers/dbms/BinaryPbIndexer/__init__.py [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+                    'unless "write_handler" is later assigned with a meaningful value'
+                )
+            else:
+                self.is_handler_loaded = True
+            return r
+
+    @property
+    def size(self) -> int:
+        """
+        The number of vectors or documents indexed.
```
Comment (cristianmtr) [inline indexers/dbms/BinaryPbIndexer/__init__.py:126]:
```suggestion
        The number of documents indexed.
```

File: indexers/dbms/BinaryPbIndexer/__init__.py [hunk 5]
```diff
@@ -0,0 +1,273 @@
...
+
+        :return: size
+        """
+        return self._size
+
+    @property
+    def default_dump_path(self):
```
Comment (cristianmtr) [inline indexers/dbms/BinaryPbIndexer/__init__.py:133]: Maybe a docstring

File: indexers/dbms/BinaryPbIndexer/binarypb.py [hunk 6]
```diff
@@ -0,0 +1,239 @@
...
+                    ],
+                )
+                for r in tmp:
+                    signature = (r[1], r[2], r[3])
+                    if np.array_equal(signature, HEADER_NONE_ENTRY):
+                        del self.header[r[0]]
+                    else:
+                        self.header[r[0]] = signature
+
+                # self.header = {
```
Comment (cristianmtr) [inline indexers/dbms/BinaryPbIndexer/binarypb.py:85]: remove

File: indexers/dbms/BinaryPbIndexer/query_compound.py [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+    def __init__(self, vector_indexer_class, kv_indexer_class, *args, **kwargs):
+        self._vec_indexer = vector_indexer_class(*args, **kwargs)
+        self._kv_indexer = kv_indexer_class(*args, **kwargs)
+
+    @requests(on='/search')
+    def search(self, docs: 'DocumentArray', parameters: Dict = None, **kwargs):
+        self._vec_indexer.search(docs, parameters)
+        kv_parameters = copy.deepcopy(parameters)
+
+        kv_parameters['traversal_paths'] = [
```
Comment (cristianmtr) [inline indexers/dbms/BinaryPbIndexer/query_compound.py:17]: so does this work?

**Review by cristianmtr — CHANGES_REQUESTED** [review]

File: indexers/dbms/BinaryPbIndexer/query_compound.py (hunk 7 above)
Comment (maximilianwerk) [reply indexers/dbms/BinaryPbIndexer/query_compound.py:17]: Yes :D

**Comment by Kelton8Z** [comment]
I wonder if this can be merged into master ?

**Comment by maximilianwerk** [comment]
> I wonder if this can be merged into master ?

No, because jina-hub has not yet changed to the 2.0 pattern.

**Pull request merged**